src/iqm/iqm_client/iqm_client.py

```diff
@@ -899,3 +911,39 @@ def _check_authentication_errors(result: requests.Response) -> None:
             raise ClientAuthenticationError('Authentication is required.')
         if result.status_code == 401:
             raise ClientAuthenticationError(f'Authentication failed: {result.text}')
+
+    def _check_not_found_error(self, response: requests.Response) -> None:
+        """Raises HTTPError with appropriate message if ``response.status_code == 404``."""
+        if response.status_code == 404:
+            version_message = ''
+            if (version_incompatibility_msg := self._check_versions()) is not None:
+                version_message = (
+                    f' This may be caused by the server version not supporting this endpoint. '
+                    f'{version_incompatibility_msg}'
+                )
+            raise HTTPError(f'{response.url} not found.{version_message}', response=response)
+
+    def _check_versions(self) -> Optional[str]:
+        """Checks the client version against compatible client versions reported by server.
+
+        Returns:
+            A message about the versions being incompatible if they are,
+            None if they are compatible or if the version information could not be obtained.
+        """
+        versions_response = requests.get(
+            self._api.url(APIEndpoint.CLIENT_LIBRARIES),
+            headers=self._default_headers(),
+            timeout=REQUESTS_TIMEOUT,
+        )
+        if versions_response.status_code == 200:
+            compatible_versions = versions_response.json()['iqm-client']
+            min_version = parse(compatible_versions['min'])
+            max_version = parse(compatible_versions['max'])
+            client_version = parse(version('iqm-client'))
+            if client_version < min_version or client_version >= max_version:
+                return (
+                    f'IQM Client version {client_version} is incompatible with the IQM server, '
+                    f'which requires {min_version} <= iqm-client < {max_version}. '
+                    f'Please use a compatible client version to make sure all features work correctly.'
+                )
```

Comment:
As it is not automatically incompatible, I think, we should let it sound less like an error message but more liek a warning (as it is implementation-wise)
```suggestion
                return (
                    f'Your IQM Client version {client_version} was built for a different version of IQM Server. '
                    f'You might encounter issues. For the best experience, consider using a version '
                    f'of IQM Client that satisfies {min_version} <= iqm-client < {max_version}.'
                )
```